src/UsgsAstroLsSensorModel.cpp

```diff
@@ -1632,14 +1662,48 @@ UsgsAstroLsSensorModel::getValidImageRange() const
 csm::EcefVector UsgsAstroLsSensorModel::getIlluminationDirection(
    const csm::EcefCoord& groundPt) const
 {
-  MESSAGE_LOG(m_logger, "Accessing illimination direction of ground point"
-              "{} {} {}."
-              "Illimination direction is not supported, throwing exception",
+  MESSAGE_LOG(m_logger, "Accessing illumination direction of ground point"
+              "{} {} {}.",
               groundPt.x, groundPt.y, groundPt.z);
-   throw csm::Error(
-      csm::Error::UNSUPPORTED_FUNCTION,
-      "Unsupported function",
-      "UsgsAstroLsSensorModel::getIlluminationDirection");
+
+  csm::ImageCoord imagePt = groundToImage(groundPt);
+  double imageTime = getImageTime(imagePt);
+
+  short numSunPositions = m_sunPosition.size();
+  short numSunVelocities = m_sunVelocity.size();
+  double x;
+  double y;
+  double z;

```

Comment:
You can just make an EcefVector here and set its x, y, z instead of creating these.